dll/win32/syssetup/wizard.c

```diff
@@ -355,11 +355,170 @@ AckPageDlgProc(HWND hwndDlg,
+    //
+    //switch (nOption)
+    //{
+    //    case PLATFORM_OPTION_2003_SERVER:
+    //        /* write ProductSuite */
+    //        pData = L"Enterprise\0Terminal Server\0";
+    //        cbData = (lstrlenA("Enterprise") + lstrlenA("Terminal Server") + 3) * sizeof(WCHAR);
+    //        error = RegSetValueExW(hKey, L"ProductSuite", 0, REG_MULTI_SZ, (BYTE *)pData, cbData);
+    //        if (error)
+    //            break;
+    //
+    //        /* write ProductType */
+    //        pData = L"ServerNT";
+    //        cbData = (lstrlenW(pData) + 1) * sizeof(WCHAR);
+    //        error = RegSetValueExW(hKey, L"ProductType", 0, REG_SZ, (BYTE *)pData, cbData);
+    //        break;
+    //
+    //    case PLATFORM_OPTION_XP:
+    //        /* write ProductSuite */
+    //        pData = L"\0";
+    //        cbData = 2 * sizeof(WCHAR);
+    //        error = RegSetValueExW(hKey, L"ProductSuite", 0, REG_MULTI_SZ, (BYTE *)pData, cbData);
+    //        if (error)
+    //            break;
+    //
+    //        /* write ProductType */
+    //        pData = L"WinNT";
+    //        cbData = (lstrlenW(pData) + 1) * sizeof(WCHAR);
+    //        error = RegSetValueExW(hKey, L"ProductType", 0, REG_SZ, (BYTE *)pData, cbData);
+    //        break;
+    //}
+    //
+    //RegCloseKey(hKey);
+    //return error == ERROR_SUCCESS;
+}
+
+static void
+OnChoose2003Server(HWND hwndDlg)
+{
+    WCHAR szText[128];
+
+    SetDlgItemTextW(hwndDlg, IDC_PLATFORM_SUITE, L"Enterprise/Terminal Server");
+    SetDlgItemTextW(hwndDlg, IDC_PLATFORM_TYPE, L"ServerNT");
+
+    LoadStringW(hDllInstance, IDS_PLATFORM2003SERVER, szText, _countof(szText));
+    SetDlgItemTextW(hwndDlg, IDC_PLATFORM_DESCRIPTION, szText);
+}
+
+static void
+OnChooseXP(HWND hwndDlg)
+{
+    WCHAR szText[128];
+
+    SetDlgItemTextW(hwndDlg, IDC_PLATFORM_SUITE, L"(None)");
+    SetDlgItemTextW(hwndDlg, IDC_PLATFORM_TYPE, L"WinNT");
+
+    LoadStringW(hDllInstance, IDS_PLATFORMXP, szText, _countof(szText));
+    SetDlgItemTextW(hwndDlg, IDC_PLATFORM_DESCRIPTION, szText);
+}
+
+static INT_PTR CALLBACK
+PlatformPageDlgProc(HWND hwndDlg,
+                    UINT uMsg,
+                    WPARAM wParam,
+                    LPARAM lParam)
+{
+    LPNMHDR lpnm;
+    PSETUPDATA pSetupData;
+    INT iItem;
+
+    pSetupData = (PSETUPDATA)GetWindowLongPtr(hwndDlg, DWLP_USER);
+
+    switch (uMsg)
+    {
+        case WM_INITDIALOG:
+        {
+            pSetupData = (PSETUPDATA)((LPPROPSHEETPAGE)lParam)->lParam;
+            SetWindowLongPtr(hwndDlg, DWLP_USER, (LONG_PTR)pSetupData);
+
+            SendDlgItemMessageW(hwndDlg, IDC_PLATFORM_OPTIONS, CB_ADDSTRING, 0, (LPARAM)L"2003 Server (Default)");
+            SendDlgItemMessageW(hwndDlg, IDC_PLATFORM_OPTIONS, CB_ADDSTRING, 0, (LPARAM)L"XP");
+
+            SendDlgItemMessageW(hwndDlg, IDC_PLATFORM_OPTIONS, CB_SETCURSEL, PLATFORM_OPTION_2003_SERVER, 0);
+            OnChoose2003Server(hwndDlg);
+        }
+        break;
```

Comment:
`WM_INITDIALOG` returns FALSE/TRUE upon finishing the control dialog initialization. Why break it?

Reply:
OK.